hail/src/main/scala/is/hail/expr/ir/IRBuilder.scala

```diff
@@ -179,6 +181,16 @@ object IRBuilder {
 
     def parallelize(nPartitions: Option[Int] = None): TableIR = TableParallelize(ir(Env.empty), nPartitions)
 
+    def arrayStructToDict(fields: IndexedSeq[String]): IRProxy = {
```

Comment:
Not sure how much we're trying to curate IRBuilder, but this seems like the right method to add?
```
def toDict(key: IR, value: IR)
```
and then:
```
.collect()
.toDict(_.selectFields(table.typ.key: _*),
        _.dropFieldList(fields))
```
@patrick-schultz ?

Reply:
Mine is simpler for all the use cases it has right now!

Reply:
I don't think IRBuilder needs to be super pruned down to a minimal set of primitives. The point is to make it easier to write IR, so having this as a convenience function is fine with me.

I think of IRBuilder as a higher level IR (just written in a different style than our core IR) which is being lowered to the core IR. That gives it the opportunity to introduce abstractions and conveniences which don't add complexity to the lower IR layers.

However, it does strike me as odd that you overloaded `dropFields` to take a list, but not `selectFields`.

Reply:
I didn't need it here!